src/go/rpk/pkg/cli/registry/schema/delete.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package schema
+
+import (
+	"fmt"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/schemaregistry"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+type deleteResponse struct {
+	Subject string `json:"subject" yaml:"subject"`
+	Version string `json:"version" yaml:"version"`
+}
+
+func newDeleteCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var sversion string
+	var permanent bool
+	cmd := &cobra.Command{
+		Use:   "delete SUBJECT --version {version}",
+		Short: "Delete a specific schema for the given subject",
+		Args:  cobra.ExactArgs(1),
+
+		Run: func(cmd *cobra.Command, args []string) {
+			f := p.Formatter
+			if h, ok := f.Help(deleteResponse{}); ok {
+				out.Exit(h)
+			}
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := schemaregistry.NewSRClient(fs, p)
+			out.MaybeDie(err, "unable to initialize schema registry client: %v", err)
+
+			version, err := parseVersion(sversion)
+			out.MaybeDieErr(err)
+
+			subject := args[0]
+			err = cl.DeleteSchema(cmd.Context(), subject, version, sr.DeleteHow(permanent))
+			out.MaybeDieErr(err)
+			if isText, _, s, err := f.Format(deleteResponse{subject, sversion}); !isText {
+				out.MaybeDie(err, "unable to print in the required format %q: %v", f.Kind, err)
+				out.Exit(s)
+			}
+			fmt.Printf("Successfully deleted schema. Subject: %q, version: %q\n", subject, sversion)
+		},
+	}
+
+	cmd.Flags().StringVar(&sversion, "version", "", "Version to delete, -1 or latest deletes the latest version")
```

Comment:
This command has a bit more help text in the version to delete info, so this might be better than my suggestion of `(latest, 0, 1...)` (and this wording also helpfully indicates that -1 deletes latest)

Reply:
Actually I prefer not mentioning `-1` for the POSIX issue we raised before so instead I went with:

```Schema version to delete (latest, 0, 1...)```